source/Cosmos.Core_Plugs/System/Collections/Generic/ComparerHelpersImpl.cs

```diff
@@ -1,117 +1,969 @@
+    }
+
+    public class NullableInt32Comparer : Comparer<Int32?>
+    {
+        public override int Compare(Int32? x, Int32? y)
+        {
+            if (x.HasValue && y.HasValue)
+            {
+                return x.Value.CompareTo(y.Value);
+            }
+
+            return 0;
+        }
+    }
+
+    public class UInt32Comparer : Comparer<UInt32>
+    {
+        public override int Compare(UInt32 x, UInt32 y)
+        {
+            return x.CompareTo(y);
+        }
+    }
+
+    public class NullableUInt32Comparer : Comparer<UInt32?>
+    {
+        public override int Compare(UInt32? x, UInt32? y)
+        {
+            if (x.HasValue && y.HasValue)
+            {
+                return x.Value.CompareTo(y.Value);
+            }
+
+            return 0;
+        }
+    }
+
+    public class Int64Comparer : Comparer<Int64>
+    {
+        public override int Compare(Int64 x, Int64 y)
+        {
+            return x.CompareTo(y);
+        }
+    }
+
+    public class NullableInt64Comparer : Comparer<Int64?>
+    {
+        public override int Compare(Int64? x, Int64? y)
+        {
+            if (x.HasValue && y.HasValue)
+            {
+                return x.Value.CompareTo(y.Value);
+            }
+
+            return 0;
+        }
+    }
+
+    public class UInt64Comparer : Comparer<UInt64>
+    {
+        public override int Compare(UInt64 x, UInt64 y)
+        {
+            return x.CompareTo(y);
+        }
+    }
+
+    public class NullableUInt64Comparer : Comparer<UInt64?>
+    {
+        public override int Compare(UInt64? x, UInt64? y)
+        {
+            if (x.HasValue && y.HasValue)
+            {
+                return x.Value.CompareTo(y.Value);
+            }
+
+            return 0;
+        }
+    }
+
+    public class GuidComparer : Comparer<Guid>
+    {
+        public override int Compare(Guid x, Guid y)
+        {
+            return x.CompareTo(y);
+        }
+    }
+
+    public class NullableGuidComparer : Comparer<Guid?>
+    {
+        public override int Compare(Guid? x, Guid? y)
+        {
+            if (x.HasValue && y.HasValue)
+            {
+                return x.Value.CompareTo(y.Value);
+            }
+
+            return 0;
+        }
+    }
+
+    public class ObjectComparer : Comparer<Object>
+    {
+        public override int Compare(Object x, Object y)
+        {
+            string text = x as string;
+            string text2 = y as string;
+            if (text != null && text2 != null)
+            {
+                return String.Compare(text, text2);
+            }
+
+            IComparable comparable = x as IComparable;
+            if (comparable != null)
+            {
+                return comparable.CompareTo(y);
+            }
+
+            IComparable comparable2 = y as IComparable;
+            if (comparable2 != null)
+            {
+                return -comparable2.CompareTo(x);
+            }
+
+            throw new NotImplementedException();

```

Comment:
Can you please implement the comparisons when the objects are null?

Reply:
Added in master.